azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/ISqlServer.java

```diff
@@ -13,7 +13,7 @@
 
     SqlServerEntity entity();
 
-    void delete();
+    ISqlServerDeleter<? extends ISqlServer> delete();
```

Comment:
void delete(resource group name, name)

Reply:
ISqlServerDeleter support both **delete by ID** and delete by Resource Group & name.

Reply:
You may get `ISqlServer` by both `id` and `rg&name`, so a simple delete method is enough
```java
Azure.az(AzureSqlServer.class).sqlServer("xxx-xx-xxx").delete()
Azure.az(AzureSqlServer.class).sqlServer("rg-xxx", "name-xxx").delete()
```